### DISCUSSION TITLE
Exclude test project containing samples from running during run all from visual studio test explorer

### DISCUSSION THREAD

**Discussion by JMolenkamp** [xunit/xunit#3097] [Question] [answered]
I tend to use test projects for sample usage of a class library, such that a specific sample is easily run. 
I obviously don't want these samples to run when running the actual tests for the library.
Is it possible to exclude an entire test project from running when running all tests from the test explorer in Visual Studio?
Something like `[Fact(Explicit = true)]`, but on a class or assembly level.

I have looked into using `FactAttribute.SkipWhen`, but I cannot find a way to determine whether tests are run outside of the sample assembly.

The way I realize this now is by giving all samples the attribute `[Fact(Explicit = true)]`. 
This alone is not enough, all samples are marked as explicit and as such, they will be run, since there are no non-explicit samples. 
A dummy test is then added with a `[Fact]` attribute without any code, such that the samples are actually excluded on run all.

**Comment by bradwilson** [comment, marked as answer]
> Is it possible to exclude an entire test project from running when running all tests from the test explorer in Visual Studio?

We have no facility for this.

> This alone is not enough, all samples are marked as explicit and as such, they will be run, since there are no non-explicit samples.

Yes, this is an unfortunate side-effect of trying to implement "a guess" about what your gesture means when running a set of tests inside Test Explorer, since it has no knowledge of what an explicit test is. We look at the test list, and if all the tests you've asked to run are explicit, then we assume you did so explicitly (pun not intended). Otherwise there would be no way to run explicit tests in Test Explorer.
